apps/vtex-my-subscriptions-3/react/components/CreationPage/FrequencySection/index.tsx

```diff
@@ -70,17 +70,25 @@ const FrequencySection: FunctionComponent<Props> = ({ frequencies }) => {
           selectedFrequency={frequencyField.value}
           onChangeFrequency={(frequency) => {
             frequencyHelper.setValue(frequency)
-            frequency.split(',') &&
-              frequency.split(',')[1] === 'WEEKLY' &&
-              purchaseHelper.setValue(WEEK_OPTIONS[new Date().getDay() - 1])
-            frequency.split(',') &&
-              (frequency.split(',')[1] === 'MONTHLY' ||
-                frequency.split(',')[1] === 'YEARLY') &&
-              purchaseHelper.setValue(
-                nextPurchaseDateField.value.getDate() <= 28
-                  ? MONTH_OPTIONS[nextPurchaseDateField.value.getDate() - 1]
+
+            const frequencyType = frequency.split(',')?.[1]
+            switch (frequencyType) {
```

Comment:
Do you think we need any default action whenever frequencyType value doesn't match those cases?

Reply:
Good point @NicholasPedroso ! But it's not necessary because the `purchaseDay` is seted as `null` before to call this function. So, this validation should be ignored and go ahead.